### DISCUSSION TITLE
Authorization

### DISCUSSION THREAD

**Discussion by TopNowchik** [thedemons/opentele#56] [Q&A] [answered]
Good day!
I want to thank you for a very useful library.
I have the following question. Why is it possible to log in to the telegram desktop without a cloud password, but when authorizing using your library, it asks for a password? I would be very grateful if you could clarify this for me.
Thanks in advance.

**Comment by thedemons** [comment, marked as answer]
> Good day!
> I want to thank you for a very useful library.
> I have the following question. Why is it possible to log in to the telegram desktop without a cloud password, but when authorizing using your library, it asks for a password? I would be very grateful if you could clarify this for me.
> Thanks in advance.

Hi, thanks for using the library, it means a lot to me.

Do you mean it throws an exception and asks for a 2FA password when loading TelegramClient from the tdata folder?
That's because when you convert TDesktop to TelegramClient, it'll re-login by default (`flag=CreateNewSession`) hence needing the password.

To actually re-use the session of that tdata, which **won't** ask for a 2FA password, you must use `flag=UseCurrentSession`.

```python
# flag=UseCurrentSession
# Convert TDesktop to Telethon using the current session.
client = await tdesk.ToTelethon(session="telethon.session", flag=UseCurrentSession)
```

You might want to read more examples and documents [here](https://opentele.readthedocs.io/en/latest/examples/convert-tdata-to-telethon/).

**Reply by TopNowchik** [reply]
You are simply the best! This is exactly what I was looking for. Thank you very much!
